### DISCUSSION TITLE
2021-12-13 Switch in branching model

### DISCUSSION THREAD

**Discussion by lauft** [GothenburgBitFactory/timewarrior#472] [Announcements]
Timewarrior will align its branching model with Taskwarrior (see https://github.com/GothenburgBitFactory/taskwarrior/discussions/2500):

* the default branch `dev` branch will be renamed to `develop`
* `master` branch will be renamed to `stable` to better reflect its contents

The old branches `master` and `dev` will stay as long as needed, but new pull requests have to be made against `develop` now.

**Comment by smemsh** [comment]
How long is "as needed?" I noticed `dev` was updated to 1.5 but `develop` has later patches.  I keep forgetting which one is the actual development branch after not paying attention for a while.  Just wondering if there is a gating condition for removal of the old branch?

To confuse things further, it seems the intention of renaming `dev` to `develop` was to track TaskWarrior's naming, but that repository's development branch is actually named `devel` :-)

**Comment by lauft** [comment]
> How long is "as needed?"

For`dev`, as long as there are [pull-requests going to `dev`](https://github.com/GothenburgBitFactory/timewarrior/pulls?q=is%3Apr+is%3Aopen+label%3Adev-branch). 

Fun fact: I actually discovered just now that one can change the target branch here in GitHub 😅, but I think those PRs should be addressed before removing `dev`.

With the release of 1.5.0, I think `master` can be removed now... 🤔 

> To confuse things further, it seems the intention of renaming dev to develop was to track TaskWarrior's naming, but that repository's development branch is actually named devel :-)

Something, somewhere went a bit wrong... 🤷🏻‍♂️ 😅
